aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -109,8 +109,26 @@ std::tuple<Tensor &,Tensor &> mode_out(Tensor& values, Tensor& indices,
            "mode only supports CPU AND CUDA backend, got: ", toString(self.type().backend()));
   dim = maybe_wrap_dim(dim, self.dim());
   if (_dimreduce_return_trivial_no_ident(values, self, dim, keepdim, "mode")) {
-    AT_ASSERT(values.dim() == 0);
-    indices.resize_({}).fill_(0);
+    if (self.dim() == 0){
+      indices.resize_({}).fill_(0);
+    } else {
+
+      AT_CHECK(self.size(dim) > 0,
+               "cannot perform reduction function mode "
+               "on tensor with no elements because the "
+               "operation does not have an identity");
+
+      std::vector<int64_t> sizes = {};
```

Comment:
I don't think you need this; as noted above, this case should be an error.

This is also incorrect in that it doesn't take `keepdim` into account.

Reply:
Again, this was done to replicate numpy's behavior. Sure, `keepdim` isn't taken into account, but my idea was to have a first feedback on the direction I was taking for the PR, and fortunately you just made it clear that basically nothing matches your plans, so, I won't be coding anything until I'm sure I got what you want.